### PR TITLE
Mantis Tower Rework

### PR DESCRIPTION
**Pull request by magid23**
In this PR:
- Remove Mantis four arm capstone and replace it with the old tier three damage upgrade, attacking with both arms. This involved removing the arms from the model, getting rid of animations, and updating various code paths.
- Added a new tier 3 mantis upgrade that temporary reduces enemy armor if they're suffering from acid.
- Added new playtests tests targeting the temporary armor reduction ability.
- Minor updates (giving all spiders a coagulation modifier).

All tests pass

### CODE REVIEW

File: Assets/Enemy/Enemy.cs [hunk 1]
```diff
@@ -301,6 +303,18 @@ public class Enemy : MonoBehaviour {
...
 
+  // Temporarily reduce the armor of this enemy by armorReduction amount. Repeated applications are
+  // handled thusly: duration can be refreshed, but does not stack. Armor power reduction does not
+  // stack either, but the greatest value is preserved.
+  public void TempReduceArmor(float armorReduction, float reductionDuration) {
+    if (TempArmorReduceEndTime < Time.time + reductionDuration) {
+      TempArmorReduceEndTime = Time.time + reductionDuration;
+    }
+    if (TempArmorReducePower < armorReduction) {
+      TempArmorReducePower = Armor < armorReduction ? Armor : armorReduction;
```
Comment (naphren) [inline Assets/Enemy/Enemy.cs:314]: how about Math.Max()

File: Assets/Enemy/Enemy.cs [hunk 2]
```diff
@@ -524,6 +538,19 @@ public class Enemy : MonoBehaviour {
     }
   }
 
+  private IEnumerator HandleTemporaryArmorReduction() {
+    while (true) {
+      yield return new WaitUntil(() => TempArmorReduceEndTime > Time.time);
```
Comment (naphren) [inline Assets/Enemy/Enemy.cs:543]: This wont handle a new armor tear happening.  It will be waiting for the last one to conclude.

**Review by naphren — CHANGES_REQUESTED** [review]

File: Assets/Enemy/Enemy.cs (hunk 1 above)
Comment (magid23) [reply Assets/Enemy/Enemy.cs:314]: It should be Math.Min, but done.

File: Assets/Enemy/Enemy.cs (hunk 2 above)
Comment (magid23) [reply Assets/Enemy/Enemy.cs:543]: This was a great point, I've adjusted the armor handling into the temporarily reduce armor method.

File: Assets/Enemy/Enemy.cs [hunk 3]
```diff
@@ -301,6 +305,19 @@ public class Enemy : MonoBehaviour {
...
+  // Temporarily reduce the armor of this enemy by armorReduction amount. Repeated applications are
+  // handled thusly: duration can be refreshed, but does not stack. Armor power reduction does not
+  // stack either, but the greatest value is preserved.
+  public void TempReduceArmor(float armorReduction, float reductionDuration) {
+    if (TempArmorReduceEndTime < Time.time + reductionDuration) {
+      TempArmorReduceEndTime = Time.time + reductionDuration;
+    }
+    if (TempArmorReducePower < armorReduction) {
+      TempArmorReducePower = Math.Min(OriginalArmor, armorReduction);
+      Armor -= TempArmorReducePower;
```
Comment (naphren) [inline Assets/Enemy/Enemy.cs:317]: Should be original armor.

File: Assets/Enemy/Enemy.cs (hunk 2 above)
Comment (naphren) [reply Assets/Enemy/Enemy.cs:548]: Still a bit of a bug, weird frame stuff but its possible we never reset armor.

**Review by naphren — CHANGES_REQUESTED** [review]

**Review by naphren — APPROVED** [review]

**Review by naphren — APPROVED** [review]

**Pull request merged**